src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs

```diff
@@ -56,23 +56,39 @@ internal static DiffEntry CreateDiffEntry(SchemaDifference difference, DiffEntry
                 // set source and target scripts
                 if (difference.SourceObject != null)
                 {
-                    string sourceScript;
-                    difference.SourceObject.TryGetScript(out sourceScript);
-                    diffEntry.SourceScript = FormatScript(sourceScript);
+                    string sourceScript = schemaComparisonResult.GetDiffEntrySourceScript(difference);
+
+                    // Child scripts that do not use alter need to be added if they are being changed, ex: "EXECUTE sp_addextendedproperty...".
+                    // Don't add scripts that start with alter because those are handled by a top level element's create
+                    // ex: if a column changes, then the parent table's script will have the column updated, but GetDiffEntrySourceScript() on the child
+                    // will return an alter table statement for updating that column when getting the child script. The child's alter script is unecessary
+                    // for displaying the script in schema compare because the comparison displays the create scripts
+                    if (!sourceScript.ToLowerInvariant().StartsWith("alter"))
```

Comment:
This check wasn't needed before because `TryGetScript()` didn't return `Alter` scripts for changes like columns that are covered in the parent's script. Now `GetDiffEntrySourceScript()` uses 2 different ways to get scripts for the differences
- database - uses `ElementScripter.Script()` which returns `Alter` scripts for some child differences
- dacpac and project - uses `service.TryGetScript() ` which doesn't return anything for those child differences

I'm not sure why they're different, but I'm guessing it has something to do with the different internal and public models in DacFx and them not being consistent, similar to inline constraints only being supported for databases with one and not the other.